packages/story-editor/src/components/floatingMenu/elements/toggleBold.js

```diff
@@ -19,18 +19,47 @@
  */
 import { Icons } from '@googleforcreators/design-system';
 import { __ } from '@googleforcreators/i18n';
+import { trackEvent } from '@googleforcreators/tracking';
+import { useCallback } from '@googleforcreators/react';
 
 /**
  * Internal dependencies
  */
-import { IconButton } from './shared';
+import useRichTextFormatting from '../../panels/design/textStyle/useRichTextFormatting';
+import updateProperties from '../../inspector/design/updateProperties';
+import { useStory } from '../../../app';
+import { IconButton, useProperties } from './shared';
 
 function ToggleBold() {
+  const { content } = useProperties(['content']);
+  const updateSelectedElements = useStory(
+    (state) => state.actions.updateSelectedElements
+  );
+
+  const pushUpdate = useCallback(
+    (update) => {
+      trackEvent('floating_menu', {
+        name: 'set_bold',
+        element: 'text',
+      });
+      updateSelectedElements({
+        properties: (element) => updateProperties(element, update, true),
+      });
+    },
+    [updateSelectedElements]
+  );
+  const {
+    textInfo: { isBold },
+    handlers: { handleClickBold },
+  } = useRichTextFormatting([{ content, type: 'text' }], pushUpdate);
```

Comment:
These are very similar in the three components, but I guess there's no good way to generalize it?

Maybe it could be generalized like:

```js
const { isToggled, toggle } = useTextToggle({
  currentValue: 'isBold',
  toggleFunction: 'handleClickBold',
  eventName: 'set_bold'),
});
```

It's not a huge gain, but I feel it makes sense here? Especially because the component here only needs the two values, `isToggled` and `toggle`, to render the button, and the rest is just internal stuff.

Reply:
Added in [2c20684](https://github.com/GoogleForCreators/web-stories-wp/pull/10875/commits/2c20684a57cae47d9ccea8f031675d1e8368f2e8)